app/models/ems_refresh/save_inventory.rb

```diff
@@ -205,17 +205,23 @@ def save_guest_devices_inventory(hardware, hashes)
     store_ids_for_new_records(hardware.guest_devices, hashes, [:device_type, :uid_ems])
   end
 
-  def save_disks_inventory(hardware, hashes)
+  def save_disks_inventory(parent, hashes)
     return if hashes.nil?
 
     # Update the associated ids
     hashes.each do |h|
-      h[:storage_id]         = h.fetch_path(:storage, :id)
+      if h[:storage_id].nil?
```

Comment:
Can this be set already? The `backing_id` and `storage_profile_id` does not do this check?